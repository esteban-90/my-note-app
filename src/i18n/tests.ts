import type { Resource } from 'i18next'
import { createInstance } from 'i18next'
import { initReactI18next } from 'react-i18next'
import { ns, supportedLngs } from './app'

const resources: Resource = {}
export const i18nInstance = createInstance()

void (async () => {
  for (const n of ns) {
    for (const lng of supportedLngs) {
      resources[lng] ??= {}
      resources[lng] = {
        ...resources[lng],
        [n]: await import(`../../public/translations/${lng}/${n}.json`),
      }
    }
  }
})()

i18nInstance.use(initReactI18next).init({
  ns,
  supportedLngs,
  fallbackLng: supportedLngs[0],
  lng: supportedLngs[0],
  react: { useSuspense: false },
  resources,
})
