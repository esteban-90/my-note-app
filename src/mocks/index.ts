import type { Note, Language, Theme } from '@/types'
import uniqid from 'uniqid'
import { makeKey } from '@/helpers'

let seconds = 0

export const makeMockNote = (content: string): Note => {
  return {
    id: uniqid('note-'),
    content,
    createdAt: new Date(1995, 11, 17, 3, 24, seconds++).toISOString(),
  }
}

const makeMockNotes = (quantity = 1): Note[] => {
  return new Array(quantity).fill(null).map((_, index) => makeMockNote(`Hello test ${index + 1}\nTesting ${index + 1}`))
}

export const mockNotes = makeMockNotes(5)

export const mockNoteKey = makeKey('notes')
export const mockLanguageKey = makeKey('language')
export const mockThemeKey = makeKey('theme')

export const mockENLanguage: Language = 'en-US'
export const mockESLanguage: Language = 'es-PY'

export const mockDayTheme: Theme = 'day'
export const mockNightTheme: Theme = 'night'

export const formatMockDate = (date: string, locale: Language = 'en-US') => {
  return new Date(date).toLocaleString(locale, { dateStyle: 'long', timeStyle: 'medium' })
}
