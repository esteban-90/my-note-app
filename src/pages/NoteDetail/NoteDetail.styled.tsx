import styled from '@emotion/styled'

export const Wrapper = styled.div`
  text-align: center;
  overflow-y: auto;
  scrollbar-width: none;

  &::-webkit-scrollbar {
    display: none;
  }
`

export const Content = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  color: var(--color-main);
  padding: 1rem;

  & svg {
    cursor: pointer;
  }
`

export const TextBox = styled.textarea`
  border: none;
  text-align: justify;
  background-color: var(--color-white);
  padding: 16px 12px;
  width: 80%;
  height: 70vh;
  resize: none;
  scrollbar-width: none;

  &:active,
  &:focus {
    outline: none;
    border: none;
  }

  &::-webkit-scrollbar {
    display: none;
  }
`
