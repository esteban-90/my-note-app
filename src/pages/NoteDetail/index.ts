export { NoteDetail as default } from './NoteDetail'
