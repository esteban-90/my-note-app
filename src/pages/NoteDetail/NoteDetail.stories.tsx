import type { ComponentStoryObj, ComponentMeta } from '@storybook/react'
import { Layout } from '@/components'
import { mockNotes } from '@/mocks'
import { NoteDetail } from './NoteDetail'

export default {
  title: 'Pages/Note Detail',
  component: NoteDetail,
  decorators: [
    (Story) => {
      return (
        <Layout>
          <Story />
        </Layout>
      )
    },
  ],
} as ComponentMeta<typeof NoteDetail>

export const NoteDetailWithoutContentExample: ComponentStoryObj<typeof NoteDetail> = {
  parameters: {
    reactRouter: {
      routePath: '/notes',
    },
  },
}

export const NoteDetailWithContentExample: ComponentStoryObj<typeof NoteDetail> = {
  parameters: {
    reactRouter: {
      routePath: '/notes/:id',
      routeParams: { id: mockNotes[0].id },
    },
  },
}
