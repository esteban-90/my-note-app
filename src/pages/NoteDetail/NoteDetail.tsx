import type { FC, ChangeEventHandler } from 'react'
import { useState, useEffect } from 'react'
import { useIntl } from 'react-intl'
import { useNavigate, useParams } from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { confirm, force } from 'notie'
import { useNotes } from '@/contexts'
import { hasBadWords } from '@/helpers'
import { NotFound } from '@/pages/NotFound'
import { Wrapper, Content, TextBox } from '@styled/NoteDetail'

export const NoteDetail: FC = () => {
  const { id = '' } = useParams()
  const { formatMessage } = useIntl()
  const navigate = useNavigate()

  const { getNote, addNote, updateNote, removeNote } = useNotes()

  const noteIsNew = !id
  const note = getNote(id)
  const noteExists = !!note

  const [noteContent, setNoteContent] = useState<string>('')

  const typingHandler: ChangeEventHandler<HTMLTextAreaElement> = (event) => {
    setNoteContent(event.target.value)
  }

  useEffect(() => {
    setNoteContent(note?.content ?? '')
  }, [])

  const addNoteText = formatMessage({ id: 'note-detail.add-note.text' })
  const updateNoteText = formatMessage({ id: 'note-detail.update-note.text' })
  const removeNoteText = formatMessage({ id: 'note-detail.remove-note.text' })

  const submitText = formatMessage({ id: 'note-detail.submit.text' })
  const cancelText = formatMessage({ id: 'note-detail.cancel.text' })
  const options = { submitText, cancelText }

  const profanityText = formatMessage({ id: 'note-detail.profanity.text' })
  const profanityConfirm = formatMessage({ id: 'note-detail.profanity.confirm' })

  const goBackTitle = formatMessage({ id: 'note-detail.go-back.title' })
  const addNoteTitle = formatMessage({ id: 'note-detail.add-note.title' })
  const removeNoteTitle = formatMessage({ id: 'note-detail.remove-note.title' })

  const warn = () => {
    force({ text: profanityText, type: 'error', buttonText: profanityConfirm }, () => {
      navigator.vibrate(100)
      navigate(-1)
    })
  }

  const add = () => {
    const newNoteContent = noteContent.trim()

    if (newNoteContent) {
      if (!hasBadWords(newNoteContent)) {
        confirm(
          { text: addNoteText, ...options },
          () => {
            navigator.vibrate(100)
            addNote(newNoteContent)
            navigate(-1)
          },
          () => navigate(-1)
        )
      } else {
        warn()
      }
    } else {
      navigate(-1)
    }
  }

  const update = () => {
    const updatedNoteContent = noteContent.trim()

    if (updatedNoteContent) {
      if (!hasBadWords(updatedNoteContent)) {
        if (noteContent !== note?.content) {
          confirm(
            { text: updateNoteText, ...options },
            () => {
              navigator.vibrate(100)
              updateNote(id, updatedNoteContent)
              navigate(-1)
            },
            () => navigate(-1)
          )
        } else {
          navigate(-1)
        }
      } else {
        warn()
      }
    } else {
      remove()
    }
  }

  const remove = () => {
    confirm({ text: removeNoteText, ...options }, () => {
      navigator.vibrate(100)
      removeNote(id)
      navigate(-1)
    })
  }

  const goBack = () => {
    if (noteIsNew) add()
    else update()
  }

  return noteIsNew || noteExists ? (
    <Wrapper>
      <Content>
        <FontAwesomeIcon icon='left-long' size='xl' onClick={goBack} title={goBackTitle} />
        <FontAwesomeIcon
          icon={noteIsNew ? 'check' : 'trash-can'}
          size='xl'
          title={noteIsNew ? addNoteTitle : removeNoteTitle}
          onClick={noteIsNew ? add : remove}
        />
      </Content>
      <TextBox aria-label='note content' value={noteContent} onChange={typingHandler} autoFocus={noteIsNew} />
    </Wrapper>
  ) : (
    <NotFound />
  )
}
