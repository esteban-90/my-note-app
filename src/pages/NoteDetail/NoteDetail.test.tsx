import { MemoryRouter, Routes, Route } from 'react-router-dom'
import { mockNotes, getNotes, saveNotes, removeNotes } from '@/mocks'
import { screen, customRender, userEvent, configure } from '@/tests'
import { NoteDetail } from './NoteDetail'

describe(NoteDetail.name, () => {
  let asFragment: () => DocumentFragment
  let UI: JSX.Element

  beforeAll(() => {
    configure({ throwSuggestions: true })
  })

  describe('if url param is not a valid id:', () => {
    beforeAll(() => {
      UI = (
        <MemoryRouter initialEntries={['/notes/test']}>
          <Routes>
            <Route path='notes'>
              <Route path=':id' element={<NoteDetail />} />
            </Route>
          </Routes>
        </MemoryRouter>
      )
    })

    beforeEach(() => {
      ;({ asFragment } = customRender(UI))
    })

    it('should match snapshot', () => {
      expect(asFragment()).toMatchSnapshot()
    })

    it('should render Not Found page', () => {
      const notFoundHeading = screen.getByRole('heading', { name: /^page not found$/i })
      expect(notFoundHeading).toBeInTheDocument()
    })
  })

  describe("if url param is a valid id but the note associated with it doesn't exist:", () => {
    beforeAll(() => {
      UI = (
        <MemoryRouter initialEntries={['/notes/note-la9l6ikp']}>
          <Routes>
            <Route path='notes'>
              <Route path=':id' element={<NoteDetail />} />
            </Route>
          </Routes>
        </MemoryRouter>
      )
    })

    beforeEach(() => {
      ;({ asFragment } = customRender(UI))
    })

    it('should match snapshot', () => {
      expect(asFragment()).toMatchSnapshot()
    })

    it('should render Not Found page', () => {
      const notFoundHeading = screen.getByRole('heading', { name: /^page not found$/i })
      expect(notFoundHeading).toBeInTheDocument()
    })
  })

  describe('creation of a new note:', () => {
    beforeAll(() => {
      UI = (
        <MemoryRouter initialEntries={['/notes']}>
          <Routes>
            <Route path='notes'>
              <Route index element={<NoteDetail />} />
            </Route>
          </Routes>
        </MemoryRouter>
      )
    })

    beforeEach(() => {
      ;({ asFragment } = customRender(UI))
    })

    it('should match snapshot', () => {
      expect(asFragment()).toMatchSnapshot()
    })

    describe('should render a link with title of:', () => {
      it('"Back to home"', () => {
        const homeLink = screen.getByText(/^back to home$/i)
        expect(homeLink).toBeInTheDocument()
      })

      it('"Add note"', () => {
        const addLink = screen.getByText(/^add note$/i)
        expect(addLink).toBeInTheDocument()
      })
    })

    it('at first, there is no content', () => {
      const textBox = screen.getByRole('textbox')
      expect(textBox).toBeInTheDocument()
      expect(textBox).toHaveValue('')
      expect(textBox).toHaveFocus()
    })

    it("should succeed if the user doesn't write bad words and add link is clicked", async () => {
      const goodNoteContent = 'Testing title\nTesting body'

      const textBox = screen.getByRole('textbox')
      await userEvent.type(textBox, goodNoteContent)

      const addLink = screen.getByText(/^add note$/i)
      await userEvent.click(addLink)

      const addAlert = await screen.findByText(/^add note\?$/i)
      expect(addAlert).toBeInTheDocument()

      const confirmButton = await screen.findByText(/^yes$/i)
      await userEvent.click(confirmButton)

      expect(navigator.vibrate).toHaveBeenCalled()

      const [{ content }] = getNotes()
      expect(content).toBe(goodNoteContent)

      removeNotes()
    })

    it("should succeed if the user doesn't write bad words and go back link is clicked", async () => {
      const goodNoteContent = 'Testing title\nTesting body'

      const textBox = screen.getByRole('textbox')
      await userEvent.type(textBox, goodNoteContent)

      const homeLink = screen.getByText(/^back to home$/i)
      await userEvent.click(homeLink)

      const addAlert = await screen.findByText(/^add note\?$/i)
      expect(addAlert).toBeInTheDocument()

      const confirmButton = await screen.findByText(/^yes$/i)
      await userEvent.click(confirmButton)

      expect(navigator.vibrate).toHaveBeenCalled()

      const [{ content }] = getNotes()
      expect(content).toBe(goodNoteContent)

      removeNotes()
    })

    it('should fail if the user writes bad words', async () => {
      const badNoteContent = 'Fucking title\nShit'

      const textBox = screen.getByRole('textbox')
      await userEvent.type(textBox, badNoteContent)

      const addLink = screen.getByText(/^add note$/i)
      await userEvent.click(addLink)

      const profanityWarn = await screen.findByText(/^profanity is not allowed!$/i)
      expect(profanityWarn).toBeInTheDocument()

      const profanityConfirm = await screen.findByText(/^ok$/i)
      await userEvent.click(profanityConfirm)

      expect(navigator.vibrate).toHaveBeenCalled()

      const notes = getNotes()
      expect(notes.find(({ content }) => content === badNoteContent)).toBeUndefined()
    })

    it("should fail if there's no content and add note link is clicked", async () => {
      const textBox = screen.getByRole('textbox')
      expect(textBox).toHaveValue('')
      expect(textBox).toHaveFocus()

      const addLink = screen.getByText(/^add note$/i)
      await userEvent.click(addLink)

      const notes = getNotes()
      expect(notes).toHaveLength(0)
    })

    it("should fail if there's already content but cancel button is clicked", async () => {
      const textBox = screen.getByRole('textbox')
      await userEvent.type(textBox, 'hello')

      const addLink = screen.getByText(/^add note$/i)
      await userEvent.click(addLink)

      const addAlert = await screen.findByText(/^add note\?$/i)
      expect(addAlert).toBeInTheDocument()

      const cancelButton = await screen.findByText(/^no$/i)
      await userEvent.click(cancelButton)

      const notes = getNotes()
      expect(notes).toHaveLength(0)
    })
  })

  describe('edition of an existing note:', () => {
    const [{ id: idToEdit, content: contentToEdit }] = mockNotes

    beforeAll(() => {
      UI = (
        <MemoryRouter initialEntries={[`/notes/${idToEdit}`]}>
          <Routes>
            <Route path='notes'>
              <Route path=':id' element={<NoteDetail />} />
            </Route>
          </Routes>
        </MemoryRouter>
      )
    })

    beforeEach(() => {
      saveNotes()
      ;({ asFragment } = customRender(UI))
    })

    afterEach(removeNotes)

    it('should match snapshot', () => {
      expect(asFragment()).toMatchSnapshot()
    })

    describe('should render a link with title of:', () => {
      it('"Back to home"', () => {
        const homeLink = screen.getByText(/^back to home$/i)
        expect(homeLink).toBeInTheDocument()
      })

      it('"Remove note"', () => {
        const removeLink = screen.getByText(/^remove note$/i)
        expect(removeLink).toBeInTheDocument()
      })
    })

    it('at first, the content should appear', () => {
      const textBox = screen.getByRole('textbox')
      expect(textBox).toBeInTheDocument()
      expect(textBox).toHaveValue(contentToEdit)
      expect(textBox).not.toHaveFocus()
    })

    it("should succeed if the user doesn't write bad words", async () => {
      const goodNoteContent = ' nice'

      const textBox = screen.getByRole('textbox')
      await userEvent.type(textBox, goodNoteContent)

      const homeLink = screen.getByText(/^back to home$/i)
      await userEvent.click(homeLink)

      const updateAlert = await screen.findByText(/^update note\?$/i)
      expect(updateAlert).toBeInTheDocument()

      const confirmButton = await screen.findByText(/^yes$/i)
      await userEvent.click(confirmButton)

      expect(navigator.vibrate).toHaveBeenCalled()

      const [{ content }] = getNotes()
      expect(content).toBe(contentToEdit + goodNoteContent)
    })

    it('should fail if the previous content is the same as the user has typed', async () => {
      const textBox = screen.getByRole('textbox')
      await userEvent.clear(textBox)
      await userEvent.type(textBox, contentToEdit)

      const homeLink = screen.getByText(/^back to home$/i)
      await userEvent.click(homeLink)

      const notes = getNotes()
      expect(notes.find(({ content }) => content === contentToEdit)).not.toBeUndefined()
    })

    it('should fail if the user writes bad words', async () => {
      const badNoteContent = ' fuck!'

      const textBox = screen.getByRole('textbox')
      await userEvent.type(textBox, badNoteContent)

      const homeLink = screen.getByText(/^back to home$/i)
      await userEvent.click(homeLink)

      const profanityWarn = await screen.findByText(/^profanity is not allowed!$/i)
      expect(profanityWarn).toBeInTheDocument()

      const profanityConfirm = await screen.findByText(/^ok$/i)
      await userEvent.click(profanityConfirm)

      expect(navigator.vibrate).toHaveBeenCalled()

      const notes = getNotes()
      expect(notes.find(({ content }) => content.includes(badNoteContent))).toBeUndefined()
    })
  })

  describe('deletion of a note:', () => {
    const [{ id: idToDelete }] = mockNotes

    beforeAll(() => {
      UI = (
        <MemoryRouter initialEntries={[`/notes/${idToDelete}`]}>
          <Routes>
            <Route path='notes'>
              <Route path=':id' element={<NoteDetail />} />
            </Route>
          </Routes>
        </MemoryRouter>
      )
    })

    beforeEach(() => {
      saveNotes()
      ;({ asFragment } = customRender(UI))
    })

    afterEach(removeNotes)

    it('should match snapshot', () => {
      expect(asFragment()).toMatchSnapshot()
    })

    it('should succeed if note exists', async () => {
      const removeLink = screen.getByText(/^remove note$/i)
      await userEvent.click(removeLink)

      const removeAlert = await screen.findByText(/^remove note\?$/i)
      expect(removeAlert).toBeInTheDocument()

      const confirmButton = await screen.findByText(/^yes$/i)
      await userEvent.click(confirmButton)

      expect(navigator.vibrate).toHaveBeenCalled()

      const noteIds = getNotes().map(({ id }) => id)
      expect(noteIds).not.toContain(idToDelete)
    })

    it('should remove the note from storage if the user has deleted all its content', async () => {
      const textBox = screen.getByRole('textbox')
      await userEvent.clear(textBox)

      const homeLink = screen.getByText(/^back to home$/i)
      await userEvent.click(homeLink)

      const removeAlert = await screen.findByText(/^remove note\?$/i)
      expect(removeAlert).toBeInTheDocument()

      const confirmButton = await screen.findByText(/^yes$/i)
      await userEvent.click(confirmButton)

      expect(navigator.vibrate).toHaveBeenCalled()

      const noteIds = getNotes().map(({ id }) => id)
      expect(noteIds).not.toContain(idToDelete)
    })
  })
})
