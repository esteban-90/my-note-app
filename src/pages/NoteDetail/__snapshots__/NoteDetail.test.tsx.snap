// Vitest Snapshot v1

exports[`NOTE DETAIL: > when the language set is English: > creation of a note: > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1ubh0lm"
  >
    <div
      class="css-qc8xmq"
    >
      <svg
        aria-labelledby="svg-inline--fa-title-91hGjc4pxNyR"
        class="svg-inline--fa fa-left-long fa-xl "
        data-icon="left-long"
        data-prefix="fas"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="svg-inline--fa-title-91hGjc4pxNyR"
        >
          Back to home
        </title>
        <path
          d="M177.5 98c-8.8-3.8-19-2-26 4.6l-144 136C2.7 243.1 0 249.4 0 256s2.7 12.9 7.5 17.4l144 136c7 6.6 17.2 8.4 26 4.6s14.5-12.5 14.5-22l0-88 288 0c17.7 0 32-14.3 32-32l0-32c0-17.7-14.3-32-32-32l-288 0 0-88c0-9.6-5.7-18.2-14.5-22z"
          fill="currentColor"
        />
      </svg>
      <svg
        aria-labelledby="svg-inline--fa-title-qtiIgZFezjqC"
        class="svg-inline--fa fa-check fa-xl "
        data-icon="check"
        data-prefix="fas"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="svg-inline--fa-title-qtiIgZFezjqC"
        >
          Add note
        </title>
        <path
          d="M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
          fill="currentColor"
        />
      </svg>
    </div>
    <textarea
      aria-label="note content"
      class="css-vhid4f"
    />
  </div>
</DocumentFragment>
`;

exports[`NOTE DETAIL: > when the language set is English: > deletion of a note: > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1ubh0lm"
  >
    <div
      class="css-qc8xmq"
    >
      <svg
        aria-labelledby="svg-inline--fa-title-BfHS4uG7uKF2"
        class="svg-inline--fa fa-left-long fa-xl "
        data-icon="left-long"
        data-prefix="fas"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="svg-inline--fa-title-BfHS4uG7uKF2"
        >
          Back to home
        </title>
        <path
          d="M177.5 98c-8.8-3.8-19-2-26 4.6l-144 136C2.7 243.1 0 249.4 0 256s2.7 12.9 7.5 17.4l144 136c7 6.6 17.2 8.4 26 4.6s14.5-12.5 14.5-22l0-88 288 0c17.7 0 32-14.3 32-32l0-32c0-17.7-14.3-32-32-32l-288 0 0-88c0-9.6-5.7-18.2-14.5-22z"
          fill="currentColor"
        />
      </svg>
      <svg
        aria-labelledby="svg-inline--fa-title-HLJutqPjdACv"
        class="svg-inline--fa fa-trash-can fa-xl "
        data-icon="trash-can"
        data-prefix="fas"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="svg-inline--fa-title-HLJutqPjdACv"
        >
          Remove note
        </title>
        <path
          d="M135.2 17.7C140.6 6.8 151.7 0 163.8 0H284.2c12.1 0 23.2 6.8 28.6 17.7L320 32h96c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 96 0 81.7 0 64S14.3 32 32 32h96l7.2-14.3zM32 128H416V448c0 35.3-28.7 64-64 64H96c-35.3 0-64-28.7-64-64V128zm96 64c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16z"
          fill="currentColor"
        />
      </svg>
    </div>
    <textarea
      aria-label="note content"
      class="css-vhid4f"
    >
      Hello test 1
Testing 1
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`NOTE DETAIL: > when the language set is English: > edition of a note: > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1ubh0lm"
  >
    <div
      class="css-qc8xmq"
    >
      <svg
        aria-labelledby="svg-inline--fa-title-ppp3lUulLCQl"
        class="svg-inline--fa fa-left-long fa-xl "
        data-icon="left-long"
        data-prefix="fas"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="svg-inline--fa-title-ppp3lUulLCQl"
        >
          Back to home
        </title>
        <path
          d="M177.5 98c-8.8-3.8-19-2-26 4.6l-144 136C2.7 243.1 0 249.4 0 256s2.7 12.9 7.5 17.4l144 136c7 6.6 17.2 8.4 26 4.6s14.5-12.5 14.5-22l0-88 288 0c17.7 0 32-14.3 32-32l0-32c0-17.7-14.3-32-32-32l-288 0 0-88c0-9.6-5.7-18.2-14.5-22z"
          fill="currentColor"
        />
      </svg>
      <svg
        aria-labelledby="svg-inline--fa-title-cIb2FX4qnCXE"
        class="svg-inline--fa fa-trash-can fa-xl "
        data-icon="trash-can"
        data-prefix="fas"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="svg-inline--fa-title-cIb2FX4qnCXE"
        >
          Remove note
        </title>
        <path
          d="M135.2 17.7C140.6 6.8 151.7 0 163.8 0H284.2c12.1 0 23.2 6.8 28.6 17.7L320 32h96c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 96 0 81.7 0 64S14.3 32 32 32h96l7.2-14.3zM32 128H416V448c0 35.3-28.7 64-64 64H96c-35.3 0-64-28.7-64-64V128zm96 64c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16z"
          fill="currentColor"
        />
      </svg>
    </div>
    <textarea
      aria-label="note content"
      class="css-vhid4f"
    >
      Hello test 1
Testing 1
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`NOTE DETAIL: > when the language set is Spanish: > creation of a note: > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1ubh0lm"
  >
    <div
      class="css-qc8xmq"
    >
      <svg
        aria-labelledby="svg-inline--fa-title-F3IOcii9ceQK"
        class="svg-inline--fa fa-left-long fa-xl "
        data-icon="left-long"
        data-prefix="fas"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="svg-inline--fa-title-F3IOcii9ceQK"
        >
          Volver a inicio
        </title>
        <path
          d="M177.5 98c-8.8-3.8-19-2-26 4.6l-144 136C2.7 243.1 0 249.4 0 256s2.7 12.9 7.5 17.4l144 136c7 6.6 17.2 8.4 26 4.6s14.5-12.5 14.5-22l0-88 288 0c17.7 0 32-14.3 32-32l0-32c0-17.7-14.3-32-32-32l-288 0 0-88c0-9.6-5.7-18.2-14.5-22z"
          fill="currentColor"
        />
      </svg>
      <svg
        aria-labelledby="svg-inline--fa-title-tkFAHgD08yq0"
        class="svg-inline--fa fa-check fa-xl "
        data-icon="check"
        data-prefix="fas"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="svg-inline--fa-title-tkFAHgD08yq0"
        >
          Agregar nota
        </title>
        <path
          d="M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
          fill="currentColor"
        />
      </svg>
    </div>
    <textarea
      aria-label="note content"
      class="css-vhid4f"
    />
  </div>
</DocumentFragment>
`;

exports[`NOTE DETAIL: > when the language set is Spanish: > deletion of a note: > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1ubh0lm"
  >
    <div
      class="css-qc8xmq"
    >
      <svg
        aria-labelledby="svg-inline--fa-title-b8g5PVVYPEkQ"
        class="svg-inline--fa fa-left-long fa-xl "
        data-icon="left-long"
        data-prefix="fas"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="svg-inline--fa-title-b8g5PVVYPEkQ"
        >
          Volver a inicio
        </title>
        <path
          d="M177.5 98c-8.8-3.8-19-2-26 4.6l-144 136C2.7 243.1 0 249.4 0 256s2.7 12.9 7.5 17.4l144 136c7 6.6 17.2 8.4 26 4.6s14.5-12.5 14.5-22l0-88 288 0c17.7 0 32-14.3 32-32l0-32c0-17.7-14.3-32-32-32l-288 0 0-88c0-9.6-5.7-18.2-14.5-22z"
          fill="currentColor"
        />
      </svg>
      <svg
        aria-labelledby="svg-inline--fa-title-f8wO7KRu6AZZ"
        class="svg-inline--fa fa-trash-can fa-xl "
        data-icon="trash-can"
        data-prefix="fas"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="svg-inline--fa-title-f8wO7KRu6AZZ"
        >
          Eliminar nota
        </title>
        <path
          d="M135.2 17.7C140.6 6.8 151.7 0 163.8 0H284.2c12.1 0 23.2 6.8 28.6 17.7L320 32h96c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 96 0 81.7 0 64S14.3 32 32 32h96l7.2-14.3zM32 128H416V448c0 35.3-28.7 64-64 64H96c-35.3 0-64-28.7-64-64V128zm96 64c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16z"
          fill="currentColor"
        />
      </svg>
    </div>
    <textarea
      aria-label="note content"
      class="css-vhid4f"
    >
      Hello test 1
Testing 1
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`NOTE DETAIL: > when the language set is Spanish: > edition of a note: > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1ubh0lm"
  >
    <div
      class="css-qc8xmq"
    >
      <svg
        aria-labelledby="svg-inline--fa-title-ZHLJak7lO6oy"
        class="svg-inline--fa fa-left-long fa-xl "
        data-icon="left-long"
        data-prefix="fas"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="svg-inline--fa-title-ZHLJak7lO6oy"
        >
          Volver a inicio
        </title>
        <path
          d="M177.5 98c-8.8-3.8-19-2-26 4.6l-144 136C2.7 243.1 0 249.4 0 256s2.7 12.9 7.5 17.4l144 136c7 6.6 17.2 8.4 26 4.6s14.5-12.5 14.5-22l0-88 288 0c17.7 0 32-14.3 32-32l0-32c0-17.7-14.3-32-32-32l-288 0 0-88c0-9.6-5.7-18.2-14.5-22z"
          fill="currentColor"
        />
      </svg>
      <svg
        aria-labelledby="svg-inline--fa-title-56vIQKhJLZdt"
        class="svg-inline--fa fa-trash-can fa-xl "
        data-icon="trash-can"
        data-prefix="fas"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="svg-inline--fa-title-56vIQKhJLZdt"
        >
          Eliminar nota
        </title>
        <path
          d="M135.2 17.7C140.6 6.8 151.7 0 163.8 0H284.2c12.1 0 23.2 6.8 28.6 17.7L320 32h96c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 96 0 81.7 0 64S14.3 32 32 32h96l7.2-14.3zM32 128H416V448c0 35.3-28.7 64-64 64H96c-35.3 0-64-28.7-64-64V128zm96 64c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16z"
          fill="currentColor"
        />
      </svg>
    </div>
    <textarea
      aria-label="note content"
      class="css-vhid4f"
    >
      Hello test 1
Testing 1
    </textarea>
  </div>
</DocumentFragment>
`;
